### PR TITLE
Prompt Persistance: Restore previous responses wherever possible

### PR DESCRIPTION
**Pull request by rwaldron**

### CODE REVIEW

**Comment by rwaldron** [comment]
The linter is failing on something that I cannot control: 

```
/home/travis/build/bocoup/threeflows/server/node_modules/aws-sdk/node_modules/uuid/package.json
  2:9  error  Parsing error: Unexpected token :
```


...Which makes me wonder why we're linting anything in node_modules. Then I looked and saw that `.eslintignore` had `/node_modules`, instead of `node_modules`. So changed that.

File: client/components/Scenario/ContentSlide.jsx [hunk 1]
```diff
@@ -41,6 +43,11 @@ class ContentSlide extends React.Component {
         const isSkip = true;
         const value = '';
 
+        if (!this.props.run) {
+            alert('Slides cannot be skipped in Preview');
```
Comment (ktbee) [inline client/components/Scenario/ContentSlide.jsx]: Do we need an alert? We didn't have one for the "Next" button in preview, I think it'll be clear enough in context why they can't skip in preview.

File: client/components/Scenario/ContentSlide.jsx [hunk 2]
```diff
@@ -62,6 +69,11 @@ class ContentSlide extends React.Component {
         const { name, value } = data;
         const { pending, required } = this.state;
 
+        if (!this.props.run) {
+            alert('Slides cannot accept responses in Preview');
```
Comment (ktbee) [inline client/components/Scenario/ContentSlide.jsx]: Likewise, not sure we need an alert here.

**Comment by rwaldron** [comment]
> Also I noticed that this doesn't fix the issue with audio response recall where the recalled audio file shows the player, but seemingly has no audio to play. Is that meant to be addressed in this PR?

It does, but it won't work with any audio that was previously recorded and saved, because the saved S3 "key" for those audio files included the literal string `"s3://v2-moments-dev/dev/"` instead of just capturing the `/audio/{run_id}/{response_id}/{user_id}/{uuid}.mpg` portion. The `"dev"` part in `"s3://v2-moments-dev/dev/"` is a prefix that we have in `process.env.ENV`, and the `"v2-moments-dev"` part is stored in `process.env.S3_BUCKET`, but neither of those parts that were previously captured can be used when actually requesting the audio file from S3 :|

This is that follow up I was telling you about, where I needed to make sure that legacy runs could be supported. I didn't implement that here because technically there shouldn't be _any_ legacy runs.

File: client/components/Scenario/ContentSlide.jsx (hunk 2 above)
Comment (rwaldron) [reply client/components/Scenario/ContentSlide.jsx]: I'd be pretty annoyed if I typed stuff into the prompts while in preview mode, hit "submit" and nothing happened.

Comment (rwaldron) [reply client/components/Scenario/ContentSlide.jsx]: Removed. But I want to come up with some kind of feedback

**Comment by rwaldron** [comment]
Rebased!!!

File: client/components/Run/index.jsx [hunk 3]
```diff
@@ -137,13 +98,16 @@ Run.propTypes = {
 };
 
 function mapStateToProps(state) {
-    const { run } = state.run;
-    return { run };
+    const { response, responses, run } = state;
```
[the remark this answers is not in the document]
Comment (rwaldron) [reply client/components/Run/index.jsx]: I'll get it cleaned up!

File: client/components/Slide/Components/ResponseRecall/Display.jsx [hunk 4]
```diff
@@ -17,6 +17,7 @@ class Display extends React.Component {
     }
 
     async componentDidMount() {
+        // TODO: replace this with call to getResponse
```
[the remark this answers is not in the document]
Comment (rwaldron) [reply client/components/Slide/Components/ResponseRecall/Display.jsx]: 😃

File: server/migrations/sql/20191203182648-correct-legacy-audio-file-keys.sql [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+)
+UPDATE run_response
+   SET response = audioresponse.response
+  FROM audioresponse
+ WHERE audioresponse.id = run_response.id;
+ALTER TABLE run_response ENABLE TRIGGER updated_at;
+COMMIT;
+-- Up above
+---
+-- Down below
```
[the remark this answers is not in the document]
Comment (rwaldron) [reply server/migrations/sql/20191203182648-correct-legacy-audio-file-keys.sql]: I thought about and I don't think it's possible: we're making a destructive change to the data, with nowhere to store a backup, but the data never worked to begin with.

File: client/components/Run/index.jsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (rwaldron) [reply client/components/Run/index.jsx]: Fixed!

**Comment by rwaldron** [comment]
Last issue addressed, merging! Thank you!

**Pull request merged**